File: Laravel/Avanzado/multitenant/app/Models/Tenant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Tenant extends Model
{
    protected $fillable = ['nombre', 'subdominio', 'activo', 'plan'];

    public function usuarios()
    {
        return $this->hasMany(User::class);
    }

    public function productos()
    {
        return $this->hasMany(Producto::class);
    }
}

File: Ruby_on_Rails/Avanzado/multitenant/app/models/tenant.rb
class Tenant < ApplicationRecord
  has_many :usuarios
  has_many :productos

  validates :nombre, :subdominio, presence: true, uniqueness: true
  validates :subdominio, format: { with: /\A[a-z0-9\-]+\z/ }

  def self.actual
    RequestStore.store[:tenant]
  end

  def self.actual=(tenant)
    RequestStore.store[:tenant] = tenant
  end
end
